test/promql/tests/test_unschedulable_pods.yaml

```diff
@@ -0,0 +1,29 @@
+# This is the main input for unit testing.
+# Only this file is passed as command line argument
+evaluation_interval: 1m
+
+rule_files:
+  - ../extracted-rules.yaml
+
+tests:
+  # Interval for the samples of the series
+  - interval: 1m
+    # Series data
+    input_series:
+
+      - series: 'kube_pod_status_unschedulable{namespace="prod", pod="prod-pod-1"}'
+        values: '1 1 1 1 1 0 0 0 0 0'
+
+      - series: 'kube_pod_status_unschedulable{namespace="prod", pod="prod-pod-2"}'
+        values: '1 1 1 1 0 0 0 0 0 0'
+
+      - series: 'kube_pod_status_unschedulable{namespace="prod-tenant", pod="prod-pod"}'
+        values: '1 1 1 1 1 1 0 0 0 0'
+
+    # Unit tests for PromQL expressions
+    promql_expr_test:
+      - expr: last_over_time(kube_pod_status_unschedulable{namespace!~".*-tenant"}[5m]) == 1
```

Comment:
By re-declaring the query used in the alerting rule here, this test no longer exercises the rule. You could, in theory, delete the rule and this test would pass. We must avoid that or else our tests won't instill any confidence. Someone will eventually forget to copy a modified alerting rule query to its respective unit tests.

We could use a recording rule to define a new metric and use that here. The downside is it adds a little more time series overhead. Any other ideas?